server/src/test/java/io/crate/expression/reference/sys/snapshot/SysSnapshotsTest.java

```diff
@@ -105,25 +102,34 @@ public void testUnavailableSnapshotsAreFilteredOut() throws Exception {
         doReturn(CompletableFuture.completedFuture(metadata))
             .when(r1)
             .getSnapshotGlobalMetadata(s3);
-        doReturn(CompletableFuture.completedFuture(new SnapshotInfo(s3, Collections.emptyList(), SnapshotState.SUCCESS)))
+        doReturn(CompletableFuture.completedFuture(new SnapshotInfo(s3, Collections.emptyList(), 1, false)))
             .when(r1)
             .getSnapshotInfo(s3);
 
         ClusterService clusterService = mock(ClusterService.class, Answers.RETURNS_DEEP_STUBS);
         when(clusterService.state().custom(SnapshotsInProgress.TYPE)).thenReturn(null);
         SysSnapshots sysSnapshots = new SysSnapshots(() -> Collections.singletonList(r1), clusterService);
-        Stream<SysSnapshot> currentSnapshots = StreamSupport.stream(
-            Spliterators.spliteratorUnknownSize(sysSnapshots.currentSnapshots().get().iterator(), Spliterator.ORDERED),
-            false
-        );
+        Iterable<SysSnapshot> iterable = sysSnapshots.currentSnapshots().get();
+        List<SysSnapshot> orderedSnapshots = new ArrayList<>();
+        for (SysSnapshot snapshot : iterable) {
+            orderedSnapshots.add(snapshot);
+        }
+        orderedSnapshots.sort(Comparator.comparing(SysSnapshot::name));
         List<String> names = new ArrayList<>();
         List<String> uuids = new ArrayList<>();
-        currentSnapshots.forEach(sysSnapshot -> {
+        List<Boolean> globalStateFlags = new ArrayList<>();
+        orderedSnapshots.forEach(sysSnapshot -> {
             names.add(sysSnapshot.name());
             uuids.add(sysSnapshot.uuid());
+            assertThat(sysSnapshot.reason()).isNull();
+            // Basic ctor used in tests, initializes total shards with 0
+            assertThat(sysSnapshot.totalShards()).isEqualTo(0);
+            globalStateFlags.add(sysSnapshot.includeGlobalState());
         });
-        assertThat(names).containsExactlyInAnyOrder("s1", "s2", "s3");
-        assertThat(uuids).containsExactlyInAnyOrder(s1.getUUID(), s2.getUUID(), s3.getUUID());
+        assertThat(names).containsExactly("s1", "s2", "s3");
+        assertThat(uuids).containsExactly(s1.getUUID(), s2.getUUID(), s3.getUUID());
+        // s1 and s2 fail on info retrieval and get undefined/null value for the flag.
+        assertThat(globalStateFlags).containsExactly(null, null, false);
```

Comment:
This test case verifies that "unavailable" snapshots are filtered out. So please do not add any other test scenarios to it but create a dedicated test instead.

Reply:
Good point, I reverted changes here and created a dedicated test